### PR TITLE
feat(maps): Add leaflet map support

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
Dependent on:
* https://github.com/hpcc-systems/Visualization/pull/3059
* https://github.com/hpcc-systems/Visualization/pull/3058
* https://github.com/hpcc-systems/Visualization/pull/3057
* https://github.com/hpcc-systems/Visualization/pull/3056

File: packages/map/src/Utility.ts [hunk 1]
```diff
@@ -343,11 +413,13 @@ export function albersUsaPr() {
     // The projections must have mutually exclusive clip regions on the sphere,
     // as this will avoid emitting interleaving lines and polygons.
     albersUsa.stream = function (stream) {
+        return cache && cacheStream === stream ? cache : cache = multiplex([lower48.stream(cacheStream = stream), alaska.stream(stream), hawaii.stream(stream)]);
```
Comment (jbrundage) [inline packages/map/src/Utility.ts]: No Puerto Rico?

File: demos/dermatology/tests/mapFactory.js [hunk 2]
```diff
@@ -506,6 +506,125 @@
...
+                            /*
+                            new CirclesLayer()
+                                .columns(DataFactory.Sample.FlightPath.columns)
+                                .data(DataFactory.Sample.FlightPath.data)
+                                .latColumn("orgin_lat")
+                                .longColumn("orgin_long")
+                                .fillColor("darkred")
+                                .strokeColor("red")
+                                .radius(500),
+                            */
```
Comment (GordonSmith) [inline demos/dermatology/tests/mapFactory.js]: Remove commented code (or remove comments)

File: packages/map/src/Utility.ts (hunk 1 above)
Comment (GordonSmith) [reply packages/map/src/Utility.ts]: Didn't it get blown away last summer?
(good catch - thx!)

File: packages/map/src/leaflet/TileLayer.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    constructor(cluster = false) {
+        super();
+        (this._layer as any).__hpcc_layer = this;
+    }
+
+    crs() {
+        return this._crs;
+    }
+
+    attibution(): string {
```
Comment (jbrundage) [inline packages/map/src/leaflet/TileLayer.ts]: Should this be "attribution"?

File: packages/map/src/leaflet/MapBox.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+declare const window: any;
+
+export class MapBoxLayer extends TileLayer {
+
+    constructor() {
+        super();
+    }
+
+    attibution(): string {
```
Comment (jbrundage) [inline packages/map/src/leaflet/MapBox.ts]: Should this be "attribution"?

File: packages/map/src/leaflet/GMap.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    init(): Promise<void> {
+        if (!this._initPromise) {
+            this._initPromise = requireGoogleMap();
+        }
+        return this._initPromise;
+    }
+
+    attibution(): string {
```
Comment (jbrundage) [inline packages/map/src/leaflet/GMap.ts]: Should this be "attribution"?

File: packages/map/src/leaflet/Leaflet.ts [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+            }
+            this._leafletMap = new Map(this._leafletElement.node(), {
+                trackResize: false,
+                zoomControl: false,
+                zoomSnap: this.mapType() === "AlbersPR" ? 0.1 : 1,
+                crs: baseLayer.crs(),
+                maxZoom: 24
+            });
+            this._leafletMap.setView([this.defaultLat(), this.defaultLong()], this.defaultZoom());
+            this._leafletMap["attributionControl"].setPrefix(baseLayer.attibution());
```
Comment (jbrundage) [inline packages/map/src/leaflet/Leaflet.ts]: Should this be "attribution"?

File: packages/map/src/leaflet/TileLayer.ts (hunk 3 above)
Comment (GordonSmith) [reply packages/map/src/leaflet/TileLayer.ts]: yes

File: packages/map/src/leaflet/MapBox.ts (hunk 4 above)
Comment (GordonSmith) [reply packages/map/src/leaflet/MapBox.ts]: yes

File: packages/map/src/leaflet/GMap.ts (hunk 5 above)
Comment (GordonSmith) [reply packages/map/src/leaflet/GMap.ts]: yes

File: packages/map/src/leaflet/Leaflet.ts (hunk 6 above)
Comment (GordonSmith) [reply packages/map/src/leaflet/Leaflet.ts]: yes

**Pull request merged**